powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/RecoveryController.java

```diff
@@ -90,6 +90,10 @@ public EciesEncryptedResponse confirmRecoveryCode(@RequestBody EciesEncryptedReq
             logger.warn("Missing nonce in ECIES request data");
             throw new PowerAuthInvalidRequestException();
         }
+        if (request.getTimestamp() == null && (!"3.0".equals(authentication.getVersion()) && !"3.1".equals(authentication.getVersion()))) {
```

Comment:
An idea to a separate issue and a pull request. We may move it to bean validator and still throw a custom exception https://stackoverflow.com/questions/62896233/how-to-throw-custom-exception-in-proper-way-when-using-javax-validation-valid Because the same code is duplicated at several places. Maybe you can add `TODO` if you wish so.